scripts/make_windows.py

```diff
@@ -1,40 +1,68 @@
+#!/usr/bin/env python3
+#
+# make_windows.py
+# Inputs: a stream of input ("-", stdin) from samtools and a threshold (integer)
+# Sums stream of input depth until cumulative coverage exceeds threshold, and makes a new bin when that occurs 
+# Outputs bins to stdout
+#
+#
+# Import modules
 import sys
 import fileinput
+# Initialize count and start variables
 count = 0
 start = 0
+# Read infile
 infile = fileinput.input(files="-")
+# Get threshold
 threshold=int(sys.argv[2])
+# Initialize chromosome as None
 chrom = None
+# Loop over input lines
 for line in infile:
     # File handling and santisation
     line = line.rstrip('\n')
     line = line.split("\t")
+    # If first iteration, set chrom to the first chromosome value
     if not chrom:
         chrom = line[0]
     # Check if new chromosome
     if(chrom != line[0]):
         # Output last chromosome, previous end (start), end of chromosome (lastknownend), count
         out = [chrom, start, lastknownend, count]
-        # Convert out to strings and join them for printing, print
+        # Convert out to strings and join them with tabs
         out = [ str(x) for x in out ]
         out = "\t".join(out)
+        # Print to stdout
         print(out)
-        # Get things ready for next loop
+        # Set new chromosome
         chrom = line[0]
+        # Set new start
         start = int(line[1])
+        # STOP THE COUNT!
         count = 0
+    # Check if all is normal and we continue with creating the bin
     if(count < threshold):
         # Add to count
         count += int(line[3])
         # Record end position of the count (in case new chromosome is skipped to)
         lastknownend = int(line[2])
+    # If count exceeds threshold
     if(count >= threshold):
```

Comment:
These sorts or comments can be considered obvious, and left out.

Better is something like 'If count exceeds threshold start a new bin' - something that explains why or what you are trying to achieve.

Those details would be less obvious to other coders.  It makes it easier for collaborators to make changes suggesting 'how' to implement code achieve the same goal